File: go/adapter/printbanner/printbanner.go
package printbanner

import (
	"github.com/hitsuji-haneta/design-pattern/go/adapter/banner"
)

type Banner interface {
	ShowWithParen() string
	ShowWithAster() string
}

type printBanner struct {
	banner Banner
}

func New(text string) *printBanner {
	return &printBanner{banner.New(text)}
}

func (pb *printBanner) PrintWeak() string {
	return pb.banner.ShowWithParen()
}

func (pb *printBanner) PrintStrong() string {
	return pb.banner.ShowWithAster()
}

File: typescript/adapter/PrintBanner.ts
import { Print } from "./interface/Print.ts";
import { Banner } from "./Banner.ts";

export class PrintBanner extends Banner implements Print {
  constructor(text: string) {
    super(text);
  }

  printWeak = (): string => {
    return this.showWithParen();
  };

  printStrong = (): string => {
    return this.showWithAster();
  };
}

